srml/contracts/src/wasm/runtime.rs

```diff
@@ -586,6 +586,86 @@ define_env!(Env, <E: Ext>,
 		Ok(())
 	},
 
+	// Attempt to restore the specified contract.
+	//
+	// This will compute a tombstone hash from the caller's storage and the given code hash and
+	// if the hash matches the hash found in the tombstone at the specified address - kill the
+	// caller contract and restore the destination contract and set the specified `rent_allowance`.
+	// All caller's funds are transfered to the destination.
+	//
+	// This function doesn't perform restoration right away but defers it to the end of the
+	// transaction. If there is no tombstone in the destination address or if the hashes don't match
+	// then restoration is cancelled and no changes are made.
```

Comment:
Perhaps make it explicit that the transaction/extrinsic is still applied if the restoration in cancelled. "no changes are made" makes it sound otherwise.

Reply:
I guess I know where the confusion comes from. The docs imply that all of this is performed right away whereas all of this is performed at the finalization stage, i.e. after all changes are commited.